### PR TITLE
Replace all errors with custom error types

### PR DESCRIPTION
**Pull request by junlarsen**

### CODE REVIEW

**Comment by junlarsen** [comment]
https://linear.app/dotkom/issue/DOT-635/establish-a-standard-way-of-handling-errors-for-the-internal-trpc-api

File: packages/core/src/modules/job-listing/job-listing-error.ts
[the remark this answers is not in the document]
Comment (henrikskog) [reply packages/core/src/modules/job-listing/job-listing-error.ts:1]: I addressed this in https://github.com/dotkom/monoweb/pull/844, lmk what you think

File: packages/core/src/modules/job-listing/job-listing-error.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+export class JobListingNotFoundError extends ApplicationError {
+  constructor(id: string) {
+    super("/problem/not-found", 404, `JobListing with ID:${id} not found`)
+  }
+}
+
+export class InvalidStartDateError extends ApplicationError {
+  constructor(reason: string) {
+    super("/problem/bad-request", 400, `The start date is invalid: ${reason}`)
```
[the remark this answers is not in the document]
Comment (henrikskog) [reply packages/core/src/modules/job-listing/job-listing-error.ts:11]: Also addressed in https://github.com/dotkom/monoweb/pull/844

**Review by henrikskog — COMMENTED** [review]
Take a look at https://github.com/dotkom/monoweb/pull/839, I went ahead and looked at some of the things I found in the PR.

Btw, should we not do something like [this](https://trpc.io/docs/server/error-handling#handling-errors) in apps/web/src/pages/api/trpc/[trpc].ts?

**Comment by junlarsen** [comment]
I will handle tRPC specific stuff in a separate PR

File: packages/core/src/error.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    public readonly status: number,
+    public readonly title: string
+  ) {
+    super(title)
+  }
+}
+
+export class IllegalStateError extends ApplicationError {
+  constructor(description: string) {
+    super("/problem/illegal-state", 500, `Illegal state reached: ${description}`)
```
Comment (henrikskog) [inline packages/core/src/error.ts:24]: According to [the rfc](https://datatracker.ietf.org/doc/html/rfc9457#name-title) the title should not change for each occurance of the error

**Review by henrikskog — CHANGES_REQUESTED** [review]
Really like this PR. What do you think about my comment?

**Pull request merged**